### PR TITLE
Smugmug Exporter

### PR DESCRIPTION
**Pull request by seehamrun**
Wire up the auth data to requests made by the smugmug exporter. This switches to using the scribe API for making oauth requests, and does some type changes and modifications to the existing SmugMugInterface.

This appears to be working as there are data models and resources filled in and passed onto the importer chosen, im still working out some kinks in the flow from smugmug -> flickr, regarding the file types, so once I get down to that cause, i'll send a followup PR, but just sending this out for review to make my future changes to the importer easier to understand. 

#362

### CODE REVIEW

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosImporter.java [hunk 1]
```diff
@@ -119,33 +141,32 @@ void importSingleAlbum(UUID jobId, String folder, PhotoAlbum inputAlbum) throws
...
-            ImmutableMap.of(
-                    "X-Smug-AlbumUri", "/api/v2/album/" + newAlbumKey,
-                    "X-Smug-ResponseType", "json",
-                    "X-Smug-Version", "v2"),
-            new TypeReference<ImageUploadResponse>() {});
-
+        "http://upload.smugmug.com/",
+        content,
+        // Headers from: https://api.smugmug.com/api/v2/doc/reference/upload.html
+        ImmutableMap.of(
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosImporter.java]: Maybe this map creation can be broken up into a different method that takes in the album key

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java [hunk 2]
```diff
@@ -34,54 +34,69 @@
...
+import org.scribe.builder.ServiceBuilder;
+import org.scribe.model.OAuthRequest;
+import org.scribe.model.Response;
+import org.scribe.model.Token;
+import org.scribe.model.Verb;
+import org.scribe.oauth.OAuthService;
 
 public class SmugMugInterface {
 
   // TODO(olsona): is this the mapper we want to use?
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java]: Let's inject a mapper passed through from the TransferExtension here

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosImporter.java (hunk 1 above)
Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugPhotosImporter.java]: Yea, I was planning on moving this with more substantial changes when switching the postRequest function to using the scribe API and signing in a followup PR. I didn't actually modify the Importer yet aside from the type changes necessary for compiling.

File: extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java (hunk 2 above)
Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/dataportabilityproject/transfer/smugmug/photos/SmugMugInterface.java]: Done.

**Pull request merged**